events/management/commands/consume_coupon_code_request_events.py

```diff
@@ -0,0 +1,172 @@
+            'basic.auth.user.info': f"{settings.SCHEMA_REGISTRY_API_KEY}:{settings.SCHEMA_REGISTRY_API_SECRET}",
+        }
+
+        schema_registry_client = SchemaRegistryClient(KAFKA_SCHEMA_REGISTRY_CONFIG)
+
+        # TODO (EventBus):
+        # 1. Reevaluate if all consumers should listen for the earliest unprocessed offset (auto.offset.reset)
+        # 2. Use Avro <-> Attr bridge to deserialize and/or throw an error if we don't know how to
+        #    deserialize messages in a particular topic. This will depend heavily on the exact API of the
+        #    Avro <-> Attr bridge, which is still under development
+
+        consumer_config = {
+            'bootstrap.servers': settings.KAFKA_BOOTSTRAP_SERVER,
+            'group.id': group_id,
+            'key.deserializer': StringDeserializer('utf-8'),
+            'value.deserializer': AvroDeserializer(schema_str=CouponCodeRequestEvent.AVRO_SCHEMA,
+                                                   schema_registry_client=schema_registry_client,
+                                                   from_dict=CouponCodeRequestEvent.from_dict),
+            'auto.offset.reset': 'earliest'
+        }
+
+        if settings.KAFKA_API_KEY and settings.KAFKA_API_SECRET:
+            consumer_config.update({
+                'sasl.mechanism': 'PLAIN',
+                'security.protocol': 'SASL_SSL',
+                'sasl.username': settings.KAFKA_API_KEY,
+                'sasl.password': settings.KAFKA_API_SECRET,
+            })
+
+        return DeserializingConsumer(consumer_config)
+
+    def handle_message(self, msg):
+        """
+        Place holder methods for how to handle an incoming message from the event bus
+        """
+        # TODO (EventBus):
+        # Rewrite this to construct and/or emit the signal eventually specified in the message.
+        coupon_code_request_dict = CouponCodeRequestEvent.to_dict(msg.value(), {})
+
+        logger.info(
+            f"Received message with key {msg.key()} and value {coupon_code_request_dict}"
+        )
+
+        try:
+            coupon_code = coupon_code_request_dict['coupon_code']
+            course_id = coupon_code_request_dict['course_id']
+            lms_user_id = coupon_code_request_dict['lms_user_id']
+            logger.info("Redeeming coupon %s for %s to enroll in %s", coupon_code, lms_user_id, course_id)
+
+            # handle redemption/fulfillment here...
+
+            # from ecommerce.coupons.api import default_code_redeemer
+            # # code, course_id, lms_user_id
+            # default_code_redeemer.redeem_coupon_code(
+            #     code=coupon_code,
+            #     course_id=course_id,
+            #     lms_user_id=lms_user_id
+            # )
+        except Exception as ex:  # pylint: disable=broad-except
+            logger.exception(ex)
+
+    def process_single_message(self, msg):
+        """
+        Handle message error or pass along for processing (separated out for easier testing)
```

Comment:
`(separated out for easier testing)` :')

i'd say remove this comment and leave it separate!